### PR TITLE
fix go test env variable invalidation

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What this PR does / why we need it**:
fix go test env variable invalidation

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
       shell: bash
       run: |
         sudo env "PATH=$PATH" bash ./build.sh
-
+    
     - name: Setup Enviroments
       run: |
         echo "PKG_CONFIG_PATH=$GITHUB_WORKSPACE/mk" >> $GITHUB_ENV
```
Comment (LiZhenCheng9527) [inline .github/workflows/main.yml:32]: save this because golangci-lint requires `api-v2-c`

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -40,4 +40,4 @@ jobs:
 
     - name: Go Test
       run: |
-        sudo env LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib:$GITHUB_WORKSPACE/api/v2-c:$GITHUB_WORKSPACE/bpf/deserialization_to_bpf_map go test -v -vet=off ./pkg/...
+        sudo env LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib:$GITHUB_WORKSPACE/api/v2-c:$GITHUB_WORKSPACE/bpf/deserialization_to_bpf_map PKG_CONFIG_PATH=$GITHUB_WORKSPACE/mk go test -v -vet=off ./pkg/...
```
Comment (hzxuzhonghu) [inline .github/workflows/main.yml:43]: isn't GITHUB_ENV a way to set global env?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -26,7 +26,7 @@ jobs:
       shell: bash
       run: |
         sudo env "PATH=$PATH" bash ./build.sh
-
+
```
Comment (hzxuzhonghu) [inline .github/workflows/main.yml:29]: restore

File: pkg/controller/workload/workload_cache_test.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
+/*
```
Comment (hzxuzhonghu) [inline pkg/controller/workload/workload_cache_test.go:1]: separate this test with another pr

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm
